fastecdsa/ecdsa.py

```diff
@@ -83,8 +89,15 @@ def verify(sig: (int, int), msg: MsgTypes, Q: Point, curve: Curve = P256, hashfu
     elif s > curve.q or s < 1:
         raise EcdsaError(
             'Invalid Signature: s is not a positive integer smaller than the curve order')
-
-    hashed = hashfunc(msg_bytes(msg)).hexdigest()
+    
+    if prehashed:
+        try:        
+            hashed = msg.hexdigest()
+        except:
+            hashed = msg.hex()
+    else:
+        hashed = hashfunc(msg_bytes(msg)).hexdigest()
+    
```

Comment:
If we assume that the result of `digest` is passed in for pre-hashed then we can just do -
```
if prehashed:
    hashed = hexlify(msg)
else:
    hashed = hashfunc(msg_bytes(msg)).hexdigest()
```